laws/src/main/scala/cats/laws/UnorderedFoldableLaws.scala

```diff
@@ -0,0 +1,46 @@
+package cats
+package laws
+
+import cats.implicits._
+import cats.kernel.CommutativeMonoid
+
+trait UnorderedFoldableLaws[F[_]] {
+  implicit def F: UnorderedFoldable[F]
+
+  def unorderedFoldConsistentWithUnorderedFoldMap[A: CommutativeMonoid](fa: F[A]): IsEq[A] =
+    F.unorderedFoldMap(fa)(identity) <-> F.unorderedFold(fa)
+
+
+
+  def forallConsistentWithExists[A](fa: F[A], p: A => Boolean): Boolean = {
+    if (F.forall(fa)(p)) {
+      val negationExists = F.exists(fa)(a => !(p(a)))
+
+      // if p is true for all elements, then there cannot be an element for which
+      // it does not hold.
+      !negationExists &&
+        // if p is true for all elements, then either there must be no elements
+        // or there must exist an element for which it is true.
+        (F.isEmpty(fa) || F.exists(fa)(p))
+    } else true // can't test much in this case
+  }
+
+  /**
+    * If `F[A]` is empty, forall must return true.
+    */
+  def forallEmpty[A](fa: F[A], p: A => Boolean): Boolean = {
+    !F.isEmpty(fa) || F.forall(fa)(p)
+  }
```

Comment:
Could it be made to be such that `existsLazy` and `forallLazy` are laws for `UnorderedFoldable` and not just `Foldable`?

Reply:
I think lazyness would require `foldRight`, no?

Reply:
I believe you can do it with `Eval`: see my duplicative [attempt](https://github.com/hrhino/cats/blob/5638c658bafcac4e06a841138d1bb5f6fa71617c/core/src/main/scala/cats/UnorderedFoldable.scala#L94-L99) at this type class.

```scala
  def exists[A](fa: F[A])(p: A => Boolean): Boolean =
    foldMapUnordered(fa)(a => Eval.later(p(a)))(new CommutativeMonoid[Eval[Boolean]] {
      def empty = Eval.False
      def combine(x: Eval[Boolean], y: Eval[Boolean]): Eval[Boolean] =
        x.flatMap(xv => if (xv) Eval.True else y)
    }).value
```

and dually for `forall`.

Reply:
You're absolutely right, this is great! Thank you so much!